platforms/f2/index.ts

```diff
@@ -0,0 +1,46 @@
+import { adEngineConfigured, bootstrapAndGetConsent, PlatformStartup } from '@platforms/shared';
+import { context } from '@wikia/ad-engine';
+import { Container } from '@wikia/dependency-injection';
+import { Action, Communicator, setupPostQuecast } from '@wikia/post-quecast';
+import { take } from 'rxjs/operators';
+import { action } from 'ts-action';
+import { ofType } from 'ts-action-operators';
+import { basicContext } from './ad-context';
+import { setupF2Ioc } from './setup-f2-ioc';
+import './styles.scss';
+
+setupPostQuecast();
+
+export const f2ready = action('[F2] Configured');
+
+function onF2Configured(): Promise<Action> {
+	const communicator = new Communicator();
+
+	return communicator.actions$.pipe(ofType(f2ready), take(1)).toPromise();
+}
+
+const load = async () => {
+	const communicator = new Communicator();
+	const f2Action = await onF2Configured();
+	const f2Payload = f2Action.payload;
+
+	context.set('src', f2Payload.src);
+	context.set('custom.namespace', f2Payload.namespace);
+	context.set('custom.adLayout', f2Payload.adLayout);
+
+	context.extend(basicContext);
+
+	const [container]: [Container, ...any[]] = await Promise.all([
+		setupF2Ioc(),
+		bootstrapAndGetConsent(),
+	]);
+	const platformStartup = container.get(PlatformStartup);
+
+	platformStartup.configure({ isMobile: f2Payload.isMobile });
+
+	communicator.dispatch(adEngineConfigured());
+
+	platformStartup.run();
+};
+
+load();
```

Comment:
I might have made mistake here, but this is how I would see it:

```suggestion

const communicator = new Communicator();

async function load(f2Config: F2Config) {
	context.set('src', f2Config.src);
	context.set('custom.namespace', f2Config.namespace);
	context.set('custom.adLayout', f2Config.adLayout);

	context.extend(basicContext);

	const [container]: [Container, ...any[]] = await Promise.all([
		setupF2Ioc(),
		bootstrapAndGetConsent(),
	]);
	const platformStartup = container.get(PlatformStartup);

	platformStartup.configure({ isMobile: f2Config.isMobile });

	communicator.dispatch(adEngineConfigured());

	platformStartup.run();
};

communicator.actions$.pipe(ofType(f2Ready), take(1)).subscribe(load);
```

Plus I am not a fan of that:
```ts
	context.set('src', f2Config.src);
	context.set('custom.namespace', f2Config.namespace);
	context.set('custom.adLayout', f2Config.adLayout);
```
In here. It should be somewhere in platform configure. Please change it. Perhaps in `F2BaseContextSetup`.